### ISSUE TITLE
Logger API compatibility broke starting in 3.6.0

### ISSUE THREAD

**Issue by garydgregory** [liquibase/liquibase#1777]
## Environment
**Liquibase Version**: 3.6.0 through 4.3.2

**Liquibase Integration & Version**:  All

**Liquibase Extension(s) & Version**: `liquibase.logging.core.AbstractLogger`

**Database Vendor & Version**: N/A

**Operating System Type & Version**: N/A

## Description
Since 3.5.5 there have been several API compatibility breaks in the logging APIs:

* 3.6 removed the buildMessage() function
* 4.0 split Logger and LoggerService apart
* 4.0 introduced the new ServiceLoader setup requirement
* Perhaps others

One place specifically that this is being felt is in the log4j-2 plugin. The goal of the plugin is to switch the logger used from the default to log4j. An example configuration is like this:

```
               <dependency>
                       <groupId>org.apache.logging.log4j</groupId>
                       <artifactId>log4j-api</artifactId>
                       <version>2.14.0</version>
                       <scope>provided</scope>
               </dependency>
               <dependency>
                       <groupId>org.apache.logging.log4j</groupId>
                       <artifactId>log4j-core</artifactId>
                       <version>2.14.0</version>
                       <scope>provided</scope>
               </dependency>
                <dependency>
                       <groupId>org.apache.logging.log4j</groupId>
                       <artifactId>log4j-liquibase</artifactId>
                       <version>2.14.0</version>
                </dependency>
                <dependency>
                       <groupId>org.liquibase</groupId>
                       <artifactId>liquibase-core</artifactId>
                       <version>4.3.1</version>
                </dependency>
```

* See https://forum.liquibase.org/t/liquibase-logging-with-log4j2/5202
* See https://github.com/apache/logging-log4j2/commits/release-2.x

From a log4j standpoint, they don't know what version of Liquibase they are running against and would like to work automatically with any version.

## Steps To Reproduce
* Clone https://github.com/apache/logging-log4j2/commits/release-2.x
* Change the Liquibase version in the root `pom.xml`
* Run `mvn clean install` to install the snapshot log4j libraries
* Watch compilation break: `mvn clean compile -pl log4j-liquibase`

## Expected/Desired Behavior
Don't break compilation.

**Comment by JonBetts** [comment]
Thanks for this Gary. As Gary wrote on this ticket, this is following on from a question I raised in the Liquibase forum regarding log4j2 integration. I subsequently opened https://github.com/liquibase/liquibase/issues/1770, but am more than happy for that to be closed as Gary's ticket has much more technical detail to help fix the issue.

**Comment by molivasdat** [comment]
HI @JonBetts If you want to close your ticket and add to the comments section your info on this one, we will have 2 scenarios that show the issue.  I especially like having a repo that shows the error with commands that are easily reproducible.  Makes it much easier to validate if we have fixed a scenario.

**Comment by nvoxland** [comment]
There was a breaking change in 3.6, but also a significant re-do of how the extensions are even loaded in 4.0. 

The large "how extensions are loaded" changes in 4.0 makes it so even if I added back the lost methods, the extension will still not be loaded without changes.

Post-large-4.0 change, we are back to being much more stringent on API breaking changes.

So, I ended up creating apache/logging-log4j2#503 which should support both 3.x and 4.x users. The fact that 3.x and 4.x loads the classes differently allows me to have separate classes that work with either 3.x or 4.x.
